genomics/genomics.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package genomics provides methods for connecting to the Genomics service.
+package genomics
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	genomicsv1 "google.golang.org/api/genomics/v1alpha2"
```

Comment:
If you are going to rename them, rename them with their versions (v1alpha2, v2alpha1)

I'd rather not see v1/v2 code here though - I would like to keep the per-version code out of the common code as much as possible.

For example, it would be reasonable to have the version specific code do something like

service, err := genomics.New(robusthttp.NewClient(ctx, genomics.GenomicsScope, basePath)

Is the only thing preventing that here is modifying the BasePath on the returned Service structure?  I don't have a good solution for that unfortunately.  We could use reflection but that's a big hammer.

Reply:
I'm not sure what you're asking me to do here.  Do you want me to move NewService/NewServiceV1 out of here and into the respective main.go?  That would be fine.  I moved them both here because I had originally planned to have export handle both versions so I wanted to share the service creation but it didn't seem worthwhile in the end.

I'm not sure what you mean in the last paragraph.  Yes, BasePath is an issue but genomics.New and genomicsv1.New are different functions that return different types so they could never be shared.